frontend/src/App.vue

```diff
@@ -186,58 +186,65 @@ export default {
-            "default"
-          ]);
-          if (question.answer === undefined) {
-            question.answer = question.default;
-          }
-        }
-        if (question._message === FUNCTION) {
-          question.message = await this.rpc.invoke("evaluateMethod", [
-            [newAnswers],
-            question.name,
-            "message"
-          ]);
-        }
-        if (question._choices === FUNCTION) {
-          question.choices = await this.rpc.invoke("evaluateMethod", [
+      try {
+        if (question.when === FUNCTION) {
+          question.isWhen = await this.rpc.invoke("evaluateMethod", [
             [newAnswers],
             question.name,
-            "choices"
+            "when"
           ]);
         }
-        if (question.validate === FUNCTION) {
-          const response = await this.rpc.invoke("evaluateMethod", [
-            [question.answer, newAnswers],
-            question.name,
-            "validate"
-          ]);
-          question.isValid = _.isString(response) ? false : response;
-          question.validationMessage = _.isString(response)
-            ? response
-            : undefined;
+
+        if (question.isWhen === true) {
+          if (question.filter === FUNCTION) {
+            question.answer = await this.rpc.invoke("evaluateMethod", [
+              [question.answer],
+              question.name,
+              "filter"
+            ]);
+          }
+          if (question._default === FUNCTION) {
+            question.default = await this.rpc.invoke("evaluateMethod", [
+              [newAnswers],
+              question.name,
+              "default"
+            ]);
+            if (question.answer === undefined) {
+              question.answer = question.default;
+            }
+          }
+          if (question._message === FUNCTION) {
+            question.message = await this.rpc.invoke("evaluateMethod", [
+              [newAnswers],
+              question.name,
+              "message"
+            ]);
+          }
+          if (question._choices === FUNCTION) {
+            question.choices = await this.rpc.invoke("evaluateMethod", [
+              [newAnswers],
+              question.name,
+              "choices"
+            ]);
+          }
+          if (question.validate === FUNCTION) {
+            const response = await this.rpc.invoke("evaluateMethod", [
+              [question.answer, newAnswers],
+              question.name,
+              "validate"
+            ]);
+            question.isValid = _.isString(response) ? false : response;
+            question.validationMessage = _.isString(response)
+              ? response
+              : undefined;
+          }
         }
+      } catch (error) {
+        question.doNotShow = true; 
+        const message = `'${question.name}' question update of generator ${this.generatorName} has failed: ${_.get(error, "message", error)}`;
+        this.rpc.invoke("logMessage", [message]);
```

Comment:
what if this question answer is important for continuing the flow?
what if user didn't open the 'console'? how would he know that an exception happen?